hunting/__main__.py

```diff
@@ -0,0 +1,278 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+import json
+import textwrap
+from dataclasses import asdict
+from pathlib import Path
+
+import click
+from tabulate import tabulate
+
+from detection_rules.misc import parse_user_config
+
+from .definitions import HUNTING_DIR
+from .markdown import MarkdownGenerator
+from .run import QueryRunner
+from .search import QueryIndex
+from .utils import (filter_elasticsearch_params, get_hunt_path, load_all_toml,
+                    load_toml, update_index_yml)
+
+
+@click.group()
+def hunting():
+    """Commands for managing hunting queries and converting TOML to Markdown."""
+    pass
+
+
+@hunting.command('generate-markdown')
+@click.argument('path', required=False)
+def generate_markdown(path: click.Path):
```

Comment:
type hint nitpick: is `path` is optional, the type should be `path: click.Path | None`. This would allow pyright to check the risky access to the var in the function.

on another note -- I found it a bit easier to convert paths directly into `pathlib.Path`, with this config for `click` -- https://github.com/elastic/react/blob/b22db38b245286d3842c074c7088257c26685ef7/react/main.py#L61-L66

Reply:
Addressed!